### PR TITLE
refactor : migrate loan account withdraw screen to compose

### LINKED ISSUE #2568: Refactor: Migration of Loan Withdraw Screen to compose
**Summary:**
Migration of Loan Withdraw Screen to compose

### PR DESCRIPTION
**Pull request by AvneetSingh2001**
Fixes #2568 

Please Add Screenshots If there are any UI changes.
| XML Screen (before)                                                                                               |  Compose Screen (Now)                                                                                            | 
|-------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------|
| screenshot_20240408_052440_mifos_mobile_720 | screenshot_20240408_055625_mifos_mobile_720 |

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawFragment.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                MifosMobileTheme {
+                    LoanAccountWithdrawScreen(
+                        withdraw = { viewModel.withdrawLoanAccount(it) },
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawFragment.kt]: We don't need to expose `withdraw`

Comment (therajanmaurya) [reply app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawFragment.kt]: Keep in LoanAccountWithdrawScreen

File: app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    ViewModel() {
+
+    private val _loanUiState = MutableStateFlow<LoanAccountWithdrawUiState>(LoanAccountWithdrawUiState.WithdrawUiReady)
+    val loanUiState: StateFlow<LoanAccountWithdrawUiState> = _loanUiState
+
+    val loanWithAssociations: StateFlow<LoanWithAssociations?> get() = _loanWithAssociations
+    private var _loanWithAssociations: MutableStateFlow<LoanWithAssociations?> = MutableStateFlow(null)
+
+    fun withdrawLoanAccount(loanReason: String?) {
+        val loanWithdraw = LoanWithdraw()
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawViewModel.kt]: use `apply` 
val loanWithdraw = LoanWithdraw().apply {
        note = loanReason
       withdrawnOnDate = DateHelper.getDateAsStringFromLong(System.currentTimeMillis())
}

File: app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawScreen.kt [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                is LoanAccountWithdrawUiState.Success -> {
+                    Toast.makeText(context, R.string.loan_application_withdrawn_successfully, Toast.LENGTH_SHORT).show()
+                    navigateBack()
+                }
+
+                is LoanAccountWithdrawUiState.Error -> {
+                    ErrorComponent(errorToast = stringResource(id = uiState.messageId))
+                }
+
+                is LoanAccountWithdrawUiState.WithdrawUiReady -> Unit
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawScreen.kt:106]: We can remove this state seems useless/.

File: app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        }
+    }
+
+    fun setLoanContent(loanWithAssociations: LoanWithAssociations?) {
+        _loanWithAssociations.value = loanWithAssociations
+    }
+}
+
+sealed class LoanAccountWithdrawUiState {
+    data object WithdrawUiReady: LoanAccountWithdrawUiState()
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawViewModel.kt:52]: Remove this state

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawScreen.kt (hunk 3 above)
Comment (AvneetSingh2001) [reply app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawScreen.kt:106]: I am using this state as a default one in the MutableStateFlow. What to keep as the default state if I removed this? Loading, Success or Error will be called only after withdraw button pressed.

**Pull request merged**